File: 1st_week/01_02_find_max_occured_alphabet.py
def find_max_occurred_alphabet(string):
    alphabet_occurrence_array = [0] * 26
    for char in string:
        if char.isalpha():
            index = ord(char) - ord("a")
            alphabet_occurrence_array[index] += 1


    max_occurrence_index = 0
    for occurrence in range(0, len(alphabet_occurrence_array)):
        if alphabet_occurrence_array[occurrence] > alphabet_occurrence_array[max_occurrence_index]:
            max_occurrence_index = occurrence
    return chr(max_occurrence_index + 97)


result = find_max_occurred_alphabet
print("정답 = i 현재 풀이 값 =", result("hello my name is dingcodingco"))
print("정답 = e 현재 풀이 값 =", result("we love algorithm"))
print("정답 = b 현재 풀이 값 =", result("best of best youtube"))

File: 1st_week/01_02_find_max-occured_alphabet.js
function find_max_occurred_alphabet(string) {
  let array = Array(26).fill(0);
  for (let i = 0; i < string.length; i++) {
    if (isAlpha(string[i])) {
      array[string[i].charCodeAt(0) - "a".charCodeAt(0)] += 1;
    }
  }

  let index = 0;
  let max_occured_alphabet_index = array[0];
  for (let i = 0; i < array.length; i++) {
    if (max_occured_alphabet_index < array[i]) {
      max_occured_alphabet_index = array[i];
      index = i;
    }
  }

  return String.fromCharCode(index + "a".charCodeAt(0));
}

function isAlpha(char) {
  return /^[A-Z]$/i.test(char);
}

result = find_max_occurred_alphabet;
console.log("정답 = i 현재 풀이 값 =", result("hello my name is dingcodingco"));
console.log("정답 = e 현재 풀이 값 =", result("we love algorithm"));
console.log("정답 = b 현재 풀이 값 =", result("best of best youtube"));
